src/components/ShopButton/ShopButton.js

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+
+import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart';
+
+import Modal from '../Modal/Modal';
+import Order from '../Order/Order';
+
+const ShopButton = (props) => {
+  const [purchasing, setPurchasing] = useState(false);
+
+  const purchaseHandler = () => {
+    setPurchasing((prev) => !prev);
+  };
+  const purchaseCancelHandler = () => {
+    setPurchasing(false);
+  };
+  const purchaseContinueHandler = () => {
+    alert('you continue!');
+  };
+
+  return (
+    <>
+      <AddShoppingCartIcon onClick={() => purchaseHandler()} />
```

Comment:
Here you are using icons directly from material Icons and I saw that you have used the <i /> tag above so there should be cancel button in Material UI icons